pkg/frontend/querymiddleware/codec_json_test.go

```diff
@@ -231,6 +232,26 @@ func TestPrometheusCodec_JSONResponse_Metrics(t *testing.T) {
 	}
 }
 
+// requireEqualHttpResponse checks the responses are the same with special handling for the Body.
+func requireEqualHttpResponse(t *testing.T, expected, actual *http.Response) {
+	// Compare all HTTP response fields except the Body
+	require.Equal(t, expected.StatusCode, actual.StatusCode)
+	require.Equal(t, expected.Header, actual.Header)
+	require.Equal(t, expected.ContentLength, actual.ContentLength)
+
+	// Verify that body types match
+	require.Equal(t, reflect.TypeOf(expected.Body), reflect.TypeOf(actual.Body))
```

Comment:
Why do we need to compare the body types? Can't we just confirm they contain the same or equivalent content?

Reply:
Because we want the tests to be sure that the response has a closer or not.